### DISCUSSION TITLE
Defining refs outside setup block

### DISCUSSION THREAD

**Discussion by bhenderson** [vuejs/pinia#2523] [Help and Questions] [answered]
Hi there,

My coworker defined his store like:

```
/* myStore.ts */
const myRef = ref(false);

export const useMyStore = defineStore('my', () => ({ myRef }))
```

I haven't been able to find any guidance on if this is an issue. I can't logically think of why it would be a problem but it's curious to me that it isn't addressed anywhere I can find.

thanks,

**Comment by posva** [comment, marked as answer]
This exposes a security risk of Cross State Pollution during SSR. It can also potentially break HMR

**Reply by bhenderson** [reply]
I see, thanks for the explanation.
